src/Context/fiber/zend_observer_fiber.php

```diff
@@ -30,8 +29,16 @@
             return false;
         }
 
-        // Has to keep reference alive
-        define(__NAMESPACE__ . '\\_ffi_fibers', $fibers);
+        // @internal
+        class FFIFiberHolder
```

Comment:
- Can you please move `class FFIFiberHolder` into a dedicated file?
- Can you please protect this feature with a constant and/or env var (at least until we can functional test this in different runtimes, with different opcache settings, etc.) ?
- Can you please remove the try-catch-block around `FFI::scope`and check for `FFIFiberHolder::$fibers` instead? The chances that users preload an undocumented .h file from a package is very low anyway, which means the exception is almost always thrown.

Reply:
I've moved this code into a class, and protected it behind an env var. also removed try/catch. It turns out that setting up fibers happens much earlier than I believed (I think because it's defined in composer.json, at least I know how it gets pre-loaded now), and that's why I've set the env var in phpunit.xml - setting it in a test, or in the `--ENV--` section of a phpt is too late.